Src/main.c

```diff
@@ -1752,29 +1649,34 @@ int main(void)
 
 	
 #ifdef USE_MAKE
-    if (firmware_info.version_major != eepromBuffer[3] || firmware_info.version_minor != eepromBuffer[4]) {
-        eepromBuffer[3] = firmware_info.version_major;
-        eepromBuffer[4] = firmware_info.version_minor;
+    if (
+        firmware_info.version_major != eepromBuffer.version.major ||
+        firmware_info.version_minor != eepromBuffer.version.minor ||
+        eeprom_layout_version > eepromBuffer.eeprom_version
```

Comment:
adding in the layout version without then updating that version will cause the ESC to call saveEEpromSettings on every boot, which is dangerous as it means an erase and re-write on every boot, which makes the chances of a power off while writing quite high, leading to incorrect settings.
I think it would actually be better not to change the layout in this PR and not change the layout version. Changes to the layout version need to be done very carefully (as this example shows) and it should not be necessary for this PR. It should be done in a less complex PR where we can see the impact more clearly and find issues like this